logger.go

```diff
@@ -128,21 +129,29 @@ func (log *logger) log(lvl Level, msg string, fields []Field) {
 		return
 	}
 
+	t := time.Now().UTC()
 	temp := log.Encoder.Clone()
 	addFields(temp, fields)
 
-	entry := newEntry(lvl, msg, temp)
-	for _, hook := range log.Hooks {
-		if err := hook(entry); err != nil {
-			log.InternalError("hook", err)
+	if len(log.Hooks) > 0 {
+		entry := Entry{
+			Level:   lvl,
+			Message: msg,
+			Time:    t,
+			enc:     temp,
 		}
+		for _, hook := range log.Hooks {
+			if err := hook(&entry); err != nil {
```

Comment:
This is going to escape since the hook could do anything with this pointer. Verified by running with `-gcflags=-m`:
```
./logger.go:144: &entry escapes to heap
```